enhancements/network/configure-ovs-alternative.md

```diff
@@ -0,0 +1,498 @@
+    networkConfig: *BOND
+  - name: master-2
+    networkConfig: *BOND
+
+This reduces duplication in install-config and the chance of errors due to
+unintended differences in node config.
+
+When this mechanism is in use, a flag will be set to indicate to configure-ovs
+that it should not attempt to configure br-ex.
+
+If changes to the bridge configuration are desired on day 2 (e.g. adding a VLAN
+or changing MTU), the user will need to install Kubernetes-NMState and use it
+to apply those changes directly to br-ex.
+
+#### Variation [optional]
+
+This could be implemented almost entirely with machine-configs, with only a simple
+service on the host to apply the provided NMState files. This is the approach
+the [prototype implementation](https://docs.google.com/document/d/1Zp1J2HbVvu-v4mHc9M594JLqt6UwdKoix8xkeSzVL98/edit#heading=h.fba4j9nvp0nl)
+used since per-host configuration was not available any other way. However,
+this feels messy as it requires every node to have the configuration for every
+other node in the cluster and select the correct one for itself. It does have
+the benefit of requiring significantly less work to implement. If users are
+writing their NMState configs in machine-configs instead of via a top-level
+OpenShift API perhaps this also side-steps the concerns about API style
+compatibility between OpenShift and NMState?
+
+### API Extensions
+
+This will not directly modify the API of OpenShift. The API for this feature is
+implemented in Kubernetes-NMState, and this does not require any specific
+modifications in that project at this time.
+
+### Implementation Details/Notes/Constraints [optional]
+
+What are the caveats to the implementation? What are some important details that
+didn't come across above. Go in to as much detail as necessary here. This might
+be a good place to talk about core concepts and how they relate.
+
+### Risks and Mitigations
+
+Because this gives the deployer a great deal of control over the bridge
+configuration for OVNK, they will have the ability to configure it incorrectly,
+which may cause problems either on initial deployment or at any point after.
+We will want to ensure that must-gather is collecting enough information about
+the bridge configuration for us to determine whether it is compliant with our
+guidelines, and we must have clear instructions on what is required for br-ex.
+
+### Drawbacks
+
+OpenShift has historically avoided per-host configuration for cluster nodes.
+This is a large departure from that philosophy, but that is unavoidable.
+Certain aspects of networking (notably static IPs) are by nature host-specific
+and cannot be handled any other way.
+
+
+## Design Details
+
+### Open Questions [optional]
+
+* There has been resistance to using the NMState API in OpenShift in the past
+  because it is not compliant with the OpenShift API Guidelines. There is work
+  underway in NMState to address this, but is that mandatory for this to be
+  implemented?
```

Comment:
Could you please share some references to both statements?

Reply:
I'll add a link here, but for my own reference it first came up in this thread: https://github.com/openshift/enhancements/pull/1267#discussion_r1013320148

There is an initial PR to smooth integration with OCP here: https://github.com/nmstate/nmstate/pull/2338 It doesn't actually change any of the API yet, but I believe it was a first step in that direction.